packages/gatsby/src/utils/babel-loader.js

```diff
@@ -0,0 +1,71 @@
+const babelLoader = require(`babel-loader`)
+const path = require(`path`)
+const fs = require(`fs`)
+
+const prepareOptions = babel => {
+  const pluginBabelConfig = require(path.join(
+    process.cwd(),
+    `./.cache/babelState.json`
+  ))
+
+  const stage = fs.readFileSync(
+    path.join(process.cwd(), `./.cache/current-stage.txt`),
+    `utf8`
+  )
+
+  // Go through babel state and create config items.
+  const reduxPlugins = []
+  const reduxPresets = []
+  pluginBabelConfig[stage].plugins.forEach(plugin => {
+    reduxPlugins.push(
+      babel.createConfigItem([require.resolve(plugin.name), plugin.options], {
+        type: `plugin`,
+      })
+    )
+  })
+  pluginBabelConfig[stage].presets.forEach(preset => {
+    reduxPresets.push(
+      babel.createConfigItem([require.resolve(preset.name), preset.options], {
+        type: `preset`,
+      })
+    )
+  })
+
+  return [reduxPresets, reduxPlugins]
+}
+
+module.exports = babelLoader.custom(babel => {
+  const toReturn = {
+    // Passed the loader options.
+    customOptions(options) {
+      return {
+        loader: {
+          cacheDirectory: true,
+          babelrc: false,
```

Comment:
I don't think you'd want to set this since you have `partialConfig.hasFilesystemConfig()` explicitly checked.

Reply:
👍